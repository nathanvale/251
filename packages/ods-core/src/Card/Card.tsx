import React, { ReactNode } from "react";
import {
  BreakpointVariants,
  SpaceVariants,
  Colors,
  CardPaddingVariants,
} from "@origin-digital/ods-types";
import { Box } from "../Box/Box";

export interface CardProps {
  padding?: CardPaddingVariants;
  "data-id"?: string;
  children?: ReactNode;
  backgroundColor?: keyof Pick<Colors, "transparent" | "grey50" | "white">;
}

type PaddingValues = Partial<Record<BreakpointVariants, SpaceVariants>>;

const cardPaddingForVariant: Record<
  CardPaddingVariants,
  {
    paddingX: PaddingValues;
    paddingY: PaddingValues;
  }
> = {
  small: {
    paddingX: {
      xs: "medium",
      sm: "xlarge",
    },
    paddingY: {
      xs: "large",
      lg: "xlarge",
    },
  },
  medium: {
    paddingX: {
      xs: "medium",
      sm: "xxlarge",
    },
    paddingY: {
      xs: "large",
      sm: "xxlarge",
    },
  },
  large: {
    paddingX: {
      xs: "medium",
      sm: "xxlarge",
      md: "xxxlarge",
    },
    paddingY: {
      xs: "large",
      sm: "xxlarge",
      md: "xxxlarge",
    },
  },
};

export const Card = ({
  "data-id": dataId,
  padding = "small",
  children,
  backgroundColor = "white",
}: CardProps) => (
  <Box
    width="full"
    backgroundColor={backgroundColor}
    data-id={dataId}
    {...cardPaddingForVariant[padding]}
  >
    {children}
  </Box>
);

Card.displayName = "Card";

Card.defaultProps = {
  "data-id": "card",
  padding: "small",
  backgroundColor: "white",
};
