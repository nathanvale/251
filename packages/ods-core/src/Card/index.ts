export { Card, defaultCardProps, cardPaddingForVariant } from "./Card";
export type CardProps = import("./Card").CardProps;
