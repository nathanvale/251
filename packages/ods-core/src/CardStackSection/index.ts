export { CardStackSection } from "./CardStackSection";
export type CardStackSectionProps = import("./CardStackSection").CardStackSectionProps;
