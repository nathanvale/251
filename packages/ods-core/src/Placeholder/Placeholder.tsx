import React from "react";
import styled, { css } from "styled-components";
import { width, height, WidthProps, HeightProps } from "styled-system";
import { ResponsiveProp } from "@origin-digital/ods-types";
import { normaliseResponsiveProp } from "@origin-digital/ods-helpers";
import { Box, BoxProps } from "../Box/Box";

type PlaceholderShapeVariant = "rectangle" | "round";
export interface PlaceholderProps {
  "data-id"?: string;
  height?: ResponsiveProp<string | number>;
  label?: string;
  shape?: PlaceholderShapeVariant;
  width?: ResponsiveProp<string | number>;
}

interface ContainerProps extends WidthProps, HeightProps {
  shape: PlaceholderProps["shape"];
}

const Container = styled<Omit<BoxProps, "width" | "height">>(Box)<
  ContainerProps
>`
  ${width}
  ${height}
  border: 2px solid rgba(51, 51, 51, 0.3);
  ${p =>
    p.shape === "round"
      ? css`
          border-radius: 50%;
        `
      : undefined}}
`;

const defaultHeight = 120;
const defaultWidth = "auto";
const defaultShape = "rectangle";

export const Placeholder = ({
  "data-id": dataId,
  label,
  width = defaultWidth,
  height = defaultHeight,
  shape = defaultShape,
}: PlaceholderProps) => {
  return (
    <Container
      data-id={dataId}
      position="relative"
      overflow="hidden"
      display="flex"
      alignItems="center"
      justifyContent="center"
      width={normaliseResponsiveProp<string | number>(width)}
      height={normaliseResponsiveProp<string | number>(height)}
      shape={shape}
      backgroundColor="grey16"
    >
      {label ? (
        <Box
          paddingX="xsmall"
          paddingY="xxsmall"
          textAlign="center"
          style={{ marginTop: "2px" }}
        >
          {label}
        </Box>
      ) : (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          style={{ width: "100%", height: "100%", position: "absolute" }}
        >
          <line
            style={{ strokeWidth: "2px", stroke: "rgba(51,51,51,.1)" }}
            x1={0}
            y1={0}
            x2="100%"
            y2="100%"
          />
          <line
            style={{ strokeWidth: "2px", stroke: "rgba(51,51,51,.1)" }}
            x1="100%"
            y1={0}
            x2={0}
            y2="100%"
          />
        </svg>
      )}
    </Container>
  );
};

Placeholder.defaultProps = {
  "data-id": "placeholder",
  height: defaultHeight,
  inline: false,
  shape: defaultShape,
  width: defaultWidth,
};

Placeholder.displayName = "Placeholder";
