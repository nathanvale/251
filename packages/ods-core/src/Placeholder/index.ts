export { Placeholder } from "./Placeholder";
export type PlaceholderProps = import("./Placeholder").PlaceholderProps;
