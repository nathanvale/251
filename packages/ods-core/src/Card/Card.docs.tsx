/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "../../../../docs/src/types";
import { Placeholder } from "../Placeholder/Placeholder";
import { CardProps, Card } from "..";

export const docs: ComponentDocs<CardProps> = {
  category: "Layout",
  componentName: "Card",
  description: "Cards provide content containers with responsive padding.",
  propDescriptions: {
    padding: "The size of the responsive paddingX and paddingY of the card.",
  },
  migrationGuide: false,
  examples: [
    {
      label: "Default Card",
      description: "Be default cards have a small padding size.",
      Code: () => (
        <Card>
          <Placeholder />
        </Card>
      ),
    },
    {
      label: "Medium Padding",
      Code: () => (
        <Card padding="medium">
          <Placeholder />
        </Card>
      ),
    },
    {
      label: "Large Padding",
      Code: () => (
        <Card padding="large">
          <Placeholder />
        </Card>
      ),
    },
  ],
};
