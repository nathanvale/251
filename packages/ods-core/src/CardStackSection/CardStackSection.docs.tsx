import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Placeholder } from "../Placeholder/Placeholder";
import { Card } from "../Card/Card";
import {
  CardStackSection,
  CardStackSectionProps,
} from "../CardStackSection/CardStackSection";

export const docs: ComponentDocs<CardStackSectionProps> = {
  category: "Layout",
  componentName: "CardStackSection",
  migrationGuide: false,
  description:
    "CardStackSection is a versatile vertical card stacking component that manages responsive max-widths, distributed space and paddingY. This component should be at your root and not be nested.",
  propDescriptions: {
    variant:
      "Default for majority of cases, widget for dashboards, card-centered for standalone cards in beskpoke flows e.g. onboarding.",
    cardWidth: `Card max-width at various breakpoints with the exception of "xs" of which has a width of "100%".`,
    paddingY: `Padding size at the top and bottom of the card stack. This prop only works with the "default" variant.`,
  },
  examples: {
    default: {
      noCard: true,
      Code: () => (
        <CardStackSection>
          <Card>
            <Placeholder />
          </Card>
          <Card>
            <Placeholder />
          </Card>
        </CardStackSection>
      ),
    },
    additional: [
      {
        label: "Setting the Card Width",
        noCard: true,
        Code: () => (
          <CardStackSection cardWidth="medium">
            <Card>
              <Placeholder />
            </Card>
            <Card>
              <Placeholder />
            </Card>
          </CardStackSection>
        ),
      },
      {
        label: "Widget Card Variant",
        noCard: true,
        Code: () => (
          <CardStackSection variant="widget">
            <Card>
              <Placeholder />
            </Card>
            <Card>
              <Placeholder />
            </Card>
          </CardStackSection>
        ),
      },
      {
        label: "Card Centered Variant",
        description: `Card centered stacks are for standalone cards in beskpoke flows e.g. onboarding. Cards are vertically centered at "md" and above. A good used case is when displaying a login form.`,
        noCard: true,
        Code: () => (
          <CardStackSection variant="card-centered">
            <Card>
              <Placeholder />
            </Card>
          </CardStackSection>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Default",
      Code: () => (
        <CardStackSection>
          <Card>
            <Placeholder />
          </Card>
          <Card>
            <Placeholder />
          </Card>
        </CardStackSection>
      ),
    },
    {
      label: "Default",
      Code: () => (
        <CardStackSection>
          <Card>
            <Placeholder />
          </Card>
          <Card>
            <Placeholder />
          </Card>
        </CardStackSection>
      ),
    },
    {
      label: "Widget",
      Code: () => (
        <CardStackSection variant="widget">
          <Card>
            <Placeholder />
          </Card>
          <Card>
            <Placeholder />
          </Card>
        </CardStackSection>
      ),
    },
    {
      label: "Card Centered",
      Code: () => (
        <CardStackSection variant="card-centered">
          <Card>
            <Placeholder />
          </Card>
        </CardStackSection>
      ),
    },
  ],
};
