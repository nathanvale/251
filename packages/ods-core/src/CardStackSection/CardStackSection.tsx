import React, { ReactNode, ReactElement } from "react";
import styled, { css } from "styled-components";
import {
  marginTop,
  marginBottom,
  MarginBottomProps,
  MarginTopProps,
} from "styled-system";
import {
  CardStackSectionVariant,
  CardStackSize,
  BreakpointVariants,
  PaddingYVariants,
  SpaceVariants,
  ResponsiveProp,
  ComponentBaseProps,
} from "@origin-digital/ods-types";
import { Box, BoxProps } from "../Box";
import { Stack } from "../Stack";
import { CardProps } from "../Card";
import { DividerCardProps } from "../DividerCard";

export interface CardStackSectionProps
  extends Omit<ComponentBaseProps, "children"> {
  children:
    | ReactElement<CardProps | DividerCardProps>[]
    | ReactElement<CardProps | DividerCardProps>;
  cardWidth?: CardStackSize;
  paddingY?: PaddingYVariants;
  variant?: CardStackSectionVariant;
}

const defaultVariant = "default";
const defaultCardWidth = "small";
const defaultPaddingY = "small";

const cardSizeForVariant: Record<
  CardStackSize,
  Record<Exclude<BreakpointVariants, "xs">, number>
> = {
  small: {
    sm: 508,
    md: 448,
    lg: 448,
    xl: 538,
  },
  medium: {
    sm: 508,
    md: 568,
    lg: 608,
    xl: 728,
  },
  large: {
    sm: 508,
    md: 568,
    lg: 768,
    xl: 918,
  },
};

interface CardSectionProps {
  cardSize: CardStackSize;
}

const CardSection = styled(Box)<CardSectionProps>`
${(p) => css`
  @media (min-width: ${p.theme.breakpoints.sm}) {
    max-width: ${cardSizeForVariant[p.cardSize].sm}px;
  }
`}

${(p) => css`
  @media (min-width: ${p.theme.breakpoints.md}) {
    max-width: ${cardSizeForVariant[p.cardSize].md}px;
  }
`}

${(p) => css`
  @media (min-width: ${p.theme.breakpoints.lg}) {
    max-width: ${cardSizeForVariant[p.cardSize].lg}px;
  }
`}

${(p) => css`
  @media (min-width: ${p.theme.breakpoints.xl}) {
    max-width: ${cardSizeForVariant[p.cardSize].xl}px;
  }
`}
`;

interface ResponsiveGridProps {
  cardWidth: CardStackSize;
  hideGutter?: boolean;
  children: ReactNode;
}

const ResponsiveCardMaxWidths = ({
  hideGutter,
  children,
  cardWidth = defaultCardWidth,
}: ResponsiveGridProps) => (
  <CardSection
    display="flex"
    flexDirection="column"
    justifyContent="center"
    width="full"
    cardSize={cardWidth}
    paddingX={{
      xs: !hideGutter ? "medium" : undefined,
    }}
  >
    {children}
  </CardSection>
);

const CardCenteredStack = ({
  children,
  "data-id": dataId,
  cardWidth = defaultCardWidth,
}: CardStackSectionProps) => (
  <Box
    display="flex"
    data-id={dataId}
    flexDirection="column"
    paddingY={{ xs: "xsmall", sm: "xsmall" }}
    alignItems="center"
    justifyContent={{ md: "center" }}
    height="full"
    width="full"
  >
    <ResponsiveCardMaxWidths cardWidth={cardWidth} hideGutter>
      {children}
    </ResponsiveCardMaxWidths>
  </Box>
);

interface WidgetContainerProps extends MarginBottomProps, MarginTopProps {}

const WidgetContainer = styled<Omit<BoxProps, "marginTop" | "marginBottom">>(
  Box
)<WidgetContainerProps>`
  ${marginTop} ${marginBottom};
`;

const WidgetStack = ({
  children,
  "data-id": dataId,
  cardWidth = defaultCardWidth,
}: CardStackSectionProps) => (
  <Box width="full" data-id={dataId}>
    <Box
      backgroundColor="secondary"
      display={{ lg: "none" }}
      style={{ height: "120px" }}
      width="full"
    />
    <WidgetContainer
      display="flex"
      justifyContent="center"
      width="full"
      marginTop={{ xs: "-88px", lg: "48px" }}
      marginBottom={{ xs: "32px", lg: "48px" }}
    >
      <ResponsiveCardMaxWidths cardWidth={cardWidth}>
        <Stack space={["medium", "xlarge"]}>{children}</Stack>
      </ResponsiveCardMaxWidths>
    </WidgetContainer>
  </Box>
);

const cardPaddingYForVariant: Record<
  PaddingYVariants,
  ResponsiveProp<SpaceVariants>
> = {
  none: {
    xs: "none",
    sm: "xsmall",
    lg: "xxlarge",
  },
  small: ["xsmall", "xxlarge"],
  medium: ["xsmall", "xxxlarge"],
};

const DefaultCardStack = ({
  children,
  "data-id": dataId,
  cardWidth = defaultCardWidth,
  paddingY = defaultPaddingY,
}: CardStackSectionProps) => (
  <Box
    data-id={dataId}
    paddingY={cardPaddingYForVariant[paddingY]}
    display="flex"
    justifyContent="center"
    width="full"
  >
    <ResponsiveCardMaxWidths cardWidth={cardWidth} hideGutter>
      <Stack space={["xsmall", "xlarge"]}>{children}</Stack>
    </ResponsiveCardMaxWidths>
  </Box>
);

export const CardStackSection = ({
  children,
  "data-id": dataId,
  variant = defaultVariant,
  cardWidth = defaultCardWidth,
  paddingY = defaultPaddingY,
}: CardStackSectionProps) => {
  if (variant === "widget") {
    return (
      <WidgetStack data-id={dataId} cardWidth={cardWidth}>
        {children}
      </WidgetStack>
    );
  } else if (variant === "card-centered") {
    return (
      <CardCenteredStack data-id={dataId} cardWidth={cardWidth}>
        {children}
      </CardCenteredStack>
    );
  } else {
    return (
      <DefaultCardStack
        data-id={dataId}
        cardWidth={cardWidth}
        paddingY={paddingY}
      >
        {children}
      </DefaultCardStack>
    );
  }
};

CardStackSection.defaultProps = {
  "data-id": "card-stack-section",
  cardWidth: defaultCardWidth,
  paddingY: defaultPaddingY,
  variant: defaultVariant,
};

CardStackSection.displayName = "CardStackSection";
