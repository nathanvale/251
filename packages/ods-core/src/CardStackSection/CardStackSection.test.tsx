import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Placeholder } from "../Placeholder";
import { Card } from "../Card";
import { CardStackSection } from "./CardStackSection";

test("it renders a CardStackSection", () => {
  const { container } = render(
    <CardStackSection>
      <Card>
        <Placeholder />
      </Card>
      <Card>
        <Placeholder />
      </Card>
    </CardStackSection>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it renders a CardStackSection with custom data-id", () => {
  const { container } = render(
    <CardStackSection data-id="myId">
      <Card>
        <Placeholder />
      </Card>
      <Card>
        <Placeholder />
      </Card>
    </CardStackSection>
  );

  expect(container.firstChild).toHaveAttribute("data-id", "myId");
});
