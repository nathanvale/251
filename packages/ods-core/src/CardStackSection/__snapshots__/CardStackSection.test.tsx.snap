// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a CardStackSection 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
  padding-bottom: 8px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c4 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #FFFFFF;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 16px;
  padding-right: 16px;
}

.c7 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
}

.c6 {
  width: auto;
  min-width: 40px;
  height: 120px;
  border: 2px solid rgba(51,51,51,0.3);
}

.c3.c3.c3 > *:not(:last-child) {
  margin-bottom: 8px;
}

@media screen and (min-width:576px) {
  .c0 {
    padding-top: 8px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 8px;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    padding-top: 48px;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    padding-top: 48px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-bottom: 8px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-bottom: 8px;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    padding-bottom: 48px;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    padding-bottom: 48px;
  }
}

@media screen and (min-width:992px) {
  .c5 {
    padding-top: 32px;
  }
}

@media screen and (min-width:992px) {
  .c5 {
    padding-bottom: 32px;
  }
}

@media screen and (min-width:576px) {
  .c5 {
    padding-left: 32px;
  }
}

@media screen and (min-width:576px) {
  .c5 {
    padding-right: 32px;
  }
}

@media screen and (min-width:576px) {
  .c3.c3.c3 > *:not(:last-child) {
    margin-bottom: 8px;
  }
}

@media screen and (min-width:768px) {
  .c3.c3.c3 > *:not(:last-child) {
    margin-bottom: 8px;
  }
}

@media screen and (min-width:992px) {
  .c3.c3.c3 > *:not(:last-child) {
    margin-bottom: 32px;
  }
}

@media screen and (min-width:1200px) {
  .c3.c3.c3 > *:not(:last-child) {
    margin-bottom: 32px;
  }
}

@media (min-width:576px) {
  .c1 {
    max-width: 508px;
  }
}

@media (min-width:768px) {
  .c1 {
    max-width: 448px;
  }
}

@media (min-width:992px) {
  .c1 {
    max-width: 448px;
  }
}

@media (min-width:1200px) {
  .c1 {
    max-width: 538px;
  }
}

<div
  class="c0"
  data-id="card-stack-section"
  display="flex"
  width="full"
>
  <div
    class="c1 c2"
    display="flex"
    width="full"
  >
    <div
      class="c3 c4"
      data-id="stack"
      display="flex"
      width="full"
    >
      <div
        class="c5"
        data-id="card"
      >
        <div
          class="c6 c7"
          data-id="placeholder"
          display="flex"
          height="120"
          overflow="hidden"
          shape="rectangle"
          width="auto"
        >
          <svg
            style="width: 100%; height: 100%; position: absolute;"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
              x1="0"
              x2="100%"
              y1="0"
              y2="100%"
            />
            <line
              style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
              x1="100%"
              x2="0"
              y1="0"
              y2="100%"
            />
          </svg>
        </div>
      </div>
      <div
        class="c5"
        data-id="card"
      >
        <div
          class="c6 c7"
          data-id="placeholder"
          display="flex"
          height="120"
          overflow="hidden"
          shape="rectangle"
          width="auto"
        >
          <svg
            style="width: 100%; height: 100%; position: absolute;"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
              x1="0"
              x2="100%"
              y1="0"
              y2="100%"
            />
            <line
              style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
              x1="100%"
              x2="0"
              y1="0"
              y2="100%"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
